packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -163,8 +164,8 @@ class IOSDevice extends Device {
     // TODO(devoncarew): Handle startPaused, debugPort.
     printTrace('Building ${app.name} for $id');
 
-    // Step 1: Install the precompiled application if necessary.
-    bool buildResult = await buildIOSXcodeProject(app, buildForDevice: true);
+    // Step 1: Install the precompiled/DBC application if necessary.
+    bool buildResult = await buildIOSXcodeProject(app, mode, buildForDevice: true);
```

Comment:
We'll need to reconcile this asymmetry in how ios and android work. For android devices, building and `startApp()` are separate; for ios devices and ios simulators, `startApp()` both builds and runs the app; this is why the build mode is ignored in the android startApp implementation.

Can you open an issue for this? One solution about be to add an abstract `build()` method to `Device`, call the apk build functionality from there for android, and move the ios building functionality out of startApp for ios+ios devices.